### DISCUSSION TITLE
How can I scroll a node in to view area?

### DISCUSSION THREAD

**Discussion by amjadaliup** [bcakmakoglu/vue-flow#487] [Q&A] [answered]
I need to highlight a node on clicking a button. So I need to scroll a node in to view and then I can highlight the specific node by adding a class. But I don't know how can I scroll to a node or scroll to a specific position programically.

**Comment by bcakmakoglu** [comment, marked as answer]
Simplest way to do this is using the `fitView` action.
The `fitView` action accepts an array of node ids that it's supposed to fit around.

```ts
const { fitView } = useVueFlow()

const onClick = () => {
	 fitView({ padding: 1, nodes: ['1'] })
}
```

**Reply by bcakmakoglu** [reply]
You can also add a duration to the options object of `fitView` to do a smooth transition to the node, i.e.

```ts
const { fitView } = useVueFlow()

const onClick = () => {
	 fitView({ padding: 1, nodes: ['1'], duration: 1000 })
}
```

**Reply by amjadaliup** [reply]
@bcakmakoglu Thank you so much :)
